test/tests/meshgenerators/refine_block_generator/tests

```diff
@@ -0,0 +1,33 @@
+[Tests]
+  issues = '#18686'
+  design = 'meshgenerators/RefineBlockGenerator.md'
+
+  [generate]
+    requirement = 'The system shall support the ability to refine'
+
+    [single_block]
+      type = 'Exodiff'
+      input = 'test_single.i'
+      exodiff = 'test_single_in.e'
+      cli_args = '--mesh-only'
+      recover = false
+      detail = 'a single block within a multi-domain mesh'
+    []
+    [multiple_block]
+      type = 'Exodiff'
+      input = 'test_multi.i'
+      exodiff = 'test_multi_in.e'
+      recover = false
+      cli_args = "--mesh-only"
+      detail = 'multiple blocks within a multi-domain mesh'
+    []
+    [neighbor_refinement]
+      type = 'Exodiff'
+      input = 'test_neighbor.i'
+      exodiff = 'test_neighbor_in.e'
+      cli_args = "--mesh-only"
+      recover = false
+      detail = 'multiple blocks within a multi-domain mesh, with neighbor refinement enabled'
+    []
+  []
+[]
```

Comment:
At least `test_multi.i` and `test_neigbor.i` can be just one file and then you can pass the `enable_neighbor_refinement` parameter via `cli_args` like so:
```
cli_args = 'Mesh/refine/enable_neighbor_refinement=false'
```

You could take that one step further and even merge all 3 input files into one, but that would be optional

Reply:
he had it this way (merged) initially but in --mesh-only the file_base in output is being ignored.
We could if we turn this into a solve and not a mesh-only case. 
Please let us know what you prefer

Reply:
ok seems you can pass the name of the output file to mesh only, like this
--mesh-only "hex_2d_err.e"'
@socratesgorilla let s do that